tests/optim/core/test_loss.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+import unittest
+from contextlib import suppress
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+import captum.optim._core.loss as opt_loss
+from captum.optim._core.output_hook import AbortForwardException, ModuleOutputsHook
+from tests.helpers.basic import BaseTest, assertArraysAlmostEqual
+
+CHANNEL_ACTIVATION_0_LOSS = 1
+CHANNEL_ACTIVATION_1_LOSS = 1
+
+
+class SimpleModel(nn.Module):
+    def __init__(self):
+        super(SimpleModel, self).__init__()
+        self.layer = nn.Conv2d(1, 2, 1)
+        # Initialize weights and biases for
+        # easy reproducibility
+        self.layer.weight.data.fill_(0)
+        self.layer.bias.data.fill_(1)
+
+    def forward(self, x):
+        return self.layer(x)
+
+
+def get_loss_value(model, loss, input_shape=[1, 1, 1, 1]):
+    if hasattr(loss.target, "__iter__"):
+        hooks = ModuleOutputsHook(loss.target)
+    else:
+        hooks = ModuleOutputsHook([loss.target])
+    with suppress(AbortForwardException):
+        model(torch.zeros(*input_shape))
+    module_outputs = hooks.consume_outputs()
+    loss_value = loss(module_outputs)
```

Comment:
nit: I think that you could potentially also used `collect_activations` to replace line 31 - 37 ? 
https://github.com/pytorch/captum/blob/optim-wip/captum/optim/_utils/models.py#L220

Reply:
Yup. Also updated `collect_activations` to work on both iterables and non-iterables for the `targets` arg.